### PR TITLE
Make remove_virtual_function() consistent

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
If there is more than one possible virtual function, we cast the object
to the appropriate type in each case, but if there is only one
possibility we don't. This PR makes us cast the object when there is
only one possibility as well.

Note that just because there is only one possibility, it does not mean
that the object already has the correct type. Also, if the object is
already the correct type then the cast isn't inserted, in all cases.

### CODE REVIEW

**Review by NathanJPhillips — APPROVED** [review]

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
@owen-jones-diffblue Would it be possible to add a regression test, in particular one that fits with "Note that just because there is only one possibility, it does not mean that the object already has the correct type."?

**Comment by owen-mc-diffblue** [comment]
Sure. I will try to get around to it soon.

**Comment by owen-mc-diffblue** [comment]
I've just added a test, which I've verified fails on develop but passes on this branch.

**Comment by owen-jones-diffblue** [comment]
This PR is ready to be merged

File: regression/cbmc-java/remove_virtual_function_typecast/VirtualFunctions.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    // multiple possibilities, one needs a cast
+    a.f();
+
+    // single possibility, does not need a cast
+    b.f();
+
+    // single possibility, needs cast
+    c.f();
+  }
+}
```
Comment (tautschnig) [inline regression/cbmc-java/remove_virtual_function_typecast/VirtualFunctions.java:27]: GitHub says there is no newline at end-of-file.

**Review by tautschnig — COMMENTED** [review]
Minor comment about a newline, but more importantly: could you please extend the commit message such that it is clear even without the context of this particular PR? It's not as easy to see in the git history later on what this test is actually about.

**Review by NathanJPhillips — APPROVED** [review]

**Pull request merged**